tests/libtransmission/torrent-files-test.cc

```diff
@@ -153,14 +153,23 @@ TEST_F(TorrentFilesTest, isSubpathPortable)
         { " foo", false },
         { "foo ", false },
 
-        // reserved names
+    // reserved names and characters (platform-dependent)
+#ifdef _WIN32
         { "COM1", false },
         { "COM1.txt", false },
         { "Com1", false },
         { "com1", false },
-
-        // reserved characters
+#else
+        { "COM1", true },
+        { "COM1.txt", true },
+        { "Com1", true },
+        { "com1", true },
+#endif
+#if defined(_WIN32) || defined(__APPLE__)
```

Comment:
colon is actually supported on Apple.
```suggestion
#if defined(_WIN32)
```

Reply:
@Coeur according to https://en.wikipedia.org/wiki/Filename#Comparison_of_filename_limitations it's a reserved character in HFS and HFS+. I'm not sure how often you'll see either of those in the wild though.

Reply:
HFS+ is not uncommon, but it supports `:` on Mac OS X/macOS, this limitation is only for legacy MacOS Systems, before OS X.
Also HFS+ drives are converted to APFS when installing macOS 10.14, which is our current minimum supported OS version.
HFS is really ancient and doesn't even worth mentioning here, as support for writing to HFS was removed in Mac OS X 10.6 😄

To summarize - it's supported, at least for Transmission use cases, but it also won't hurt anybody to sanitize `:`, I think.

Reply:
I think it's reasonable to sanitize this when `:` shows up as `/` in different context, likely causing user confusion.
Can also remove it if that's where the opinion goes of course.

Reply:
Let's only have `/` as a reserved character, as this is how it is technically on macOS, no matter HFS+ or APFS.

Reply:
Done.